lib/Glimmer.pm

```diff
@@ -0,0 +1,494 @@
+# use strict;
+# use Data::Dumper;
+
+# my $out = Glimmer::call_genes_with_glimmer(file_with_path, { verbose => 1 });
+
+# print Dumper($out);
+
+package Glimmer;
+use strict;
+use File::Slurp;
+use File::Temp;
+use Data::Dumper;
+
+sub call_genes_with_glimmer
+{
+    my($contigs_file, $opts) = @_;
+
+    my $genetic_code     =   $opts->{genetic_code} || 11;   #...Default to the "standard" code
+
+    my $train            = qq();   #...Flag associated with '-train' switch
+    my $training_tbl     = qq();
+    my $training_contigs = qq();
+    
+    my $min_training_len = $opts->{min_training_len} || 2000;   #...Shortest contig used for self-training
+    my $verbose = $opts->{verbose};
+
+    my $glimmeropts = qq(-o50 -g110 -t30 -l);   #...NOTE: Make these the default for a switch
+
+    my $genetic_code_switch = qq(-z $genetic_code);
+
+    my $id_prefix = $opts->{id_prefix} || "prot.";
+
+    if (! $training_contigs) { $training_contigs = $contigs_file }
```

Comment:
`$training_contigs ||= $contigs_file;`

if `$training_contigs` does not evaluate to true, set it to `$contigs_file`